### PR TITLE
Renew

### PR DESCRIPTION
**Pull request by jefflau**

### CODE REVIEW

**Comment by makoto** [comment]
Styling looks heavily broken on narrower window.

Screenshot 2019-04-23 at 11 06 33

File: src/components/SingleName/DetailsItemEditable.js [hunk 1]
```diff
@@ -136,7 +208,20 @@ const Editable = ({
...
+  //only used with Expiration date
+  let duration
+  let expirationDate
+  const [years, setYears] = useState(1)
+  const { price: ethUsdPrice, loading: ethUsdPriceLoading } = useEthPrice(
+    keyName === 'Expiration Date'
+  )
+  if (keyName === 'Expiration Date') {
+    duration = parseFloat(years) * yearInSeconds
+    console.log(duration)
```
Comment (makoto) [inline src/components/SingleName/DetailsItemEditable.js]: Remove console.log

File: src/components/SingleName/DetailsItemEditable.js [hunk 2]
```diff
@@ -254,17 +346,25 @@ const Editable = ({
...
-
-                        mutation({
-                          variables
+                        const variables = getVariables(keyName, {
+                          domain,
+                          variableName,
+                          newValue,
+                          duration
                         })
+                        console.log(variables)
```
Comment (makoto) [inline src/components/SingleName/DetailsItemEditable.js]: Remove console.log

File: src/graphql/mutations.js [hunk 3]
```diff
@@ -110,14 +110,20 @@ export const SET_REGISTRANT = gql`
...
-    commit(label: $label, owner: $owner, secret: $secret) @client
+  mutation commit($label: String) {
+    commit(label: $label) @client
   }
 `
 
 export const REGISTER = gql`
   mutation register($label: String, $duration: Int) {
-    register(label: $label, duration: $duration, secret: $secret) @client
+    register(label: $label, duration: $duration) @client
```
Comment (makoto) [inline src/graphql/mutations.js]: Why do you take out secret when registrar.reister still takes it as an argument ? ub.com/ensdomains/ens-app/blob/renew/src/api/registrar.js#L295

File: src/graphql/mutations.js [hunk 4]
```diff
@@ -110,14 +110,20 @@ export const SET_REGISTRANT = gql`
 `
 
 export const COMMIT = gql`
-  mutation commit($label: String, $owner: String, $secret: String) {
-    commit(label: $label, owner: $owner, secret: $secret) @client
```
Comment (makoto) [inline src/graphql/mutations.js]: Same as comment below

File: src/graphql/mutations.js (hunk 3 above)
Comment (jefflau) [reply src/graphql/mutations.js]: It takes from inside the resolver, not from the front-end, so we don't need it in the mutation

File: src/graphql/mutations.js (hunk 4 above)
Comment (jefflau) [reply src/graphql/mutations.js]: Same answer as above. The secret is not taken from the UI, so it doesn't need to be part of the mutation

**Comment by jefflau** [comment]
> Styling looks heavily broken on narrower window.
> 
> Screenshot 2019-04-23 at 11 06 33

Mostly fixed this. The problem only really occurs when reducing the size of the window whilst renew is open as React spring calculates the height for us, but if you move it the content can become larger than the height it gives. We can create a more robust solution, but I've fixed it for the most part.

**Pull request merged**